build/templates/_grpc_stub_interpreter.py.mako

```diff
@@ -51,20 +50,22 @@ class GrpcStubInterpreter(object):
                         error_code = int(value)
                     except ValueError:
                         error_message += f'\nError status: {value}'
-            grpc_error = rpc_error.code()
 
-        if grpc_error == grpc.StatusCode.UNAVAILABLE:
-            error_message = 'Failed to connect to server'
-        elif grpc_error == grpc.StatusCode.UNIMPLEMENTED:
-            error_message = (
-                'This operation is not supported by the NI gRPC Device Server being used. Upgrade NI gRPC Device Server.'
-            )
-        elif grpc_error == grpc.StatusCode.NOT_FOUND:
-            raise errors.DriverTooOldError()
+            grpc_error = rpc_error.code()
+            if grpc_error == grpc.StatusCode.UNAVAILABLE:
+                error_message = 'Failed to connect to server'
+            elif grpc_error == grpc.StatusCode.UNIMPLEMENTED:
+                error_message = (
+                    'This operation is not supported by the NI gRPC Device Server being used. Upgrade NI gRPC Device Server.'
+                )
+            elif grpc_error == grpc.StatusCode.NOT_FOUND:
+                raise errors.DriverTooOldError()
+            elif grpc_error == grpc.StatusCode.INVALID_ARGUMENT:
+                raise ValueError(error_message)
+            elif error_code is None:
+                raise errors.RpcError(grpc_error, error_message)
 
-        if grpc_error == grpc.StatusCode.INVALID_ARGUMENT:
-            raise ValueError(error_message)
-        elif error_code is None:
+        if error_code is None:
```

Comment:
Now that you check it in the 'catch' block, I don't think this check is legitimate. If the 'try' block executes to completion, the `error_code` will be set, and the status from which it was set is an integer. Right?

Reply:
The problem is the UNAVAILABLE/UNIMPLEMENTED use cases above. (Which we don't have a test for, except right now while NI-Switch is failing.) They set `error_message` but leave `error_code` as `None`, and fall through to right here.

But maybe those should raise an RpcError now.

Reply:
Yeah, line 65 above should be `if` rather than `elif`.

Reply:
Done. And added unit tests.